tools/walletextension/vk_repository.go

```diff
@@ -0,0 +1,72 @@
+package walletextension
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto/ecies"
+)
+
+func NewViewingKeyRepository() *ViewingKeyRepository {
+	return &ViewingKeyRepository{
+		viewingKeysPrivate: make(map[common.Address]*ecies.PrivateKey),
+		viewingKeysPublic:  make(map[common.Address][]byte),
+	}
+}
+
+type ViewingKeyRepository struct {
+	viewingKeysPrivate map[common.Address]*ecies.PrivateKey // Maps an address to its private viewing key.
+	viewingKeysPublic  map[common.Address][]byte            // Maps an address to its public viewing key bytes.
+}
+
+func (v *ViewingKeyRepository) IsReady() bool {
+	return len(v.viewingKeysPrivate) > 0
+}
+
+func (v *ViewingKeyRepository) DecryptBytes(respBytes []byte) ([]byte, error) {
```

Comment:
I think this should take both the request and the response. We can use the `from` from the request to determine how to decrypt it